packages/components/src/components/hds/badge/index.ts

```diff
@@ -107,7 +111,7 @@ export default class HdsBadgeIndexComponent extends Component {
    * @type {string|null}
    * @default null
    */
-  get icon() {
+  get icon(): string | null {
```

Comment:
in other components these are declared inside the `types.ts` files

Reply:
Great point. I don't believe you _need_ to explicitly add the return type as it's inferred by typescript in this example.

<img width="464" alt="Screenshot 2024-03-26 at 9 48 51 AM" src="https://github.com/hashicorp/design-system/assets/5448834/70ebf58c-6ed3-4be8-938f-18c2ff937de0">

Reply:
Typescript seems to infer the return type really well when it's a basic type.  It's not as great when your return type is something like the enum values types that we are using. 

<img width="682" alt="Screenshot 2024-03-26 at 10 00 14 AM" src="https://github.com/hashicorp/design-system/assets/5448834/61e693da-bc3f-45f6-911b-39c34c241c8e">

vs.

<img width="682" alt="Screenshot 2024-03-26 at 10 00 00 AM" src="https://github.com/hashicorp/design-system/assets/5448834/83ee1ff0-a413-4307-8241-4c65af2e0462">

Reply:
my two cents here is to your point I think primitive time is easily inferred by TS on its own so it would only make sense to add return types in cases where we return complex types, otherwise it might add more noise then benefit

Reply:
agree – if they are primitive types we shouldn't need to add them – for more elaborate types (such as `HdsBadgeColors`) there are benefits, and we usually place them in `type.js` (which is how they're setup for this component as well).

this being said, @WenInCode if you have time could you please remove the returned primitive types from the getter in this component (there a few of them)? if time is tight let me know and I can add a commit for this